
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  

  # Position Controllers ---------------------------------------
  shoulder_pan_joint_position_controller:
    type: effort_controllers/JointEffortController
    joint: shoulder_pan_joint
    pid: {p: 6000.0, i: 460.0, d: 540, i_clamp_min: -100.0, i_clamp_max: 100.0}
  shoulder_lift_joint_position_controller:
    type: effort_controllers/JointEffortController
    joint: shoulder_lift_joint
    pid: {p: 6000.0, i: 460.0, d: 540, i_clamp_min: -400.0, i_clamp_max: 400.0}
  elbow_joint_position_controller:
    type: effort_controllers/JointEffortController
    joint: elbow_joint
    pid: {p: 1000.0, i: 460.0, d: 440, i_clamp_min: -100.0, i_clamp_max: 100.0}
  wrist_1_joint_position_controller:
    type: effort_controllers/JointEffortController
    joint: wrist_1_joint
    pid: {p: 200.0, i: 10.0, d: 20.0, i_clamp_min: -400.0, i_clamp_max: 400.0}
  wrist_2_joint_position_controller:
    type: effort_controllers/JointEffortController
    joint: wrist_2_joint
    pid: {p: 100.0, i: 0.1, d: 10.0, i_clamp_min: -100.0, i_clamp_max: 100.0}
  wrist_3_joint_position_controller:
    type: effort_controllers/JointEffortController
    joint: wrist_3_joint
    pid: {p: 100.0, i: 0.1, d: 10.0, i_clamp_min: -100.0, i_clamp_max: 100.0}
